tests/unit/registerEditorEventHandler.test.ts

```diff
@@ -1,115 +1,69 @@
-import { registerEditorEventHandler, useCKEditor } from '../../src';
+import { registerEditorEventHandler } from '../../src';
 
 function init() {
 	describe( 'registerEditorEventHandler', () => {
-		/**
-		 * Registers an event handler within editor.
-		 */
+		const spyOnEventOn = jasmine.createSpy( 'editor.on' );
+		const spyOnRemoveListener = jasmine.createSpy( 'editor.removeListener' );
+		const createEditor = () => ( { on: spyOnEventOn, removeListener: spyOnRemoveListener } );
+
+		afterEach( () => {
+			spyOnEventOn.calls.reset();
+			spyOnRemoveListener.calls.reset();
+		} );
+
 		it( 'registers / unregisters event handler', async () => {
-			const element = createDivElement();
 			const onInstanceReady = jasmine.createSpy( 'onInstanceReady' );
-			const { result } = renderHook( () =>
-				useCKEditor( {
-					element
-				} )
-			);
-			await waitForValueToChange( () => !!result.current.editor );
 			const unregister = registerEditorEventHandler( {
-				editor: result.current.editor,
+				editor: createEditor(),
 				evtName: 'instanceReady',
 				handler: onInstanceReady
 			} );
-			expect(
-				result.current.editor.hasListeners( 'instanceReady' )
-			).toBeTrue();
-			await waitForValueToChange( () => result.current.status === 'ready' );
-			expect( onInstanceReady ).toHaveBeenCalledTimes( 1 );
+			expect( spyOnEventOn ).toHaveBeenCalledTimes( 1 );
+			expect( spyOnEventOn ).toHaveBeenCalledWith( 'instanceReady', onInstanceReady, null, undefined, undefined );
 			unregister();
-			expect(
-				result.current.editor.hasListeners( 'instanceReady' )
-			).toBeUndefined();
+			expect( spyOnRemoveListener ).toHaveBeenCalledTimes( 1 );
+			expect( spyOnRemoveListener ).toHaveBeenCalledWith( 'instanceReady', onInstanceReady );
 		} );
 
-		/**
-		 * With debugging enabled, events are logged to console.
-		 */
 		it( 'turns on `debug` mode', async () => {
 			const windw = window as any;
-			windw.console.log = jasmine.createSpy( 'windw.console.log' );
-			const element = createDivElement();
+			const log = window.console.log;
+			windw.console.log = jasmine.createSpy( 'window.console.log' );
 			const onInstanceReady = jasmine.createSpy( 'onInstanceReady' );
-			const { result } = renderHook( () =>
-				useCKEditor( {
-					element
-				} )
-			);
-			await waitForValueToChange( () => !!result.current.editor );
 			registerEditorEventHandler( {
-				editor: result.current.editor,
+				editor: createEditor(),
 				evtName: 'instanceReady',
 				handler: onInstanceReady,
 				debug: true
 			} );
-			await waitForValueToChange( () => result.current.status === 'ready' );
-			expect( onInstanceReady ).toHaveBeenCalledTimes( 1 );
+			expect( spyOnEventOn ).toHaveBeenCalledTimes( 1 );
+			expect( spyOnEventOn ).toHaveBeenCalledWith( 'instanceReady', jasmine.any( Function ), null, undefined, undefined );
 			expect( windw.console.log ).toHaveBeenCalled();
+			window.console.log = log;
```

Comment:
Not very important, but failing assertions will prevent restoring proper window API which may impact other tests, so it may make sense to keep that in the teardown function.

Reply:
Yeah, we used to have that in `afterEach` hook but that's the only test that overrides window API. Given that tests are run sequentially it doesn't really matter where the API is restored.

Reply:
Moved to `afterEach` hook in https://github.com/ckeditor/ckeditor4-react/pull/292/commits/f9a3e253b4e4b9c1c135f1b5f4310774d8d6767c as discussed on Slack.